src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let val of this) {
+      if (val === null || val === undefined) {
+        val = '';
```

Comment:
it's not a good idea to change incoming variable from loop, you can do just result += '';